include/revng/Pipeline/ContainerInspector.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/ManagedStatic.h"
+
+#include "revng/Pipeline/Containers.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Target.h"
+
+namespace Pipeline {
+
+/// If a downstream user of reveng wishes to extend the enumeration and
+/// removal rule from a container it can do so by extending this class,
+/// specialized with the type of the container in question.
```

Comment:
* [ ] Before describing how to extend the class, briefly describe its purpose.
  Something like "An inspector enables the enumeration and removal of a specific kind from the container Container."
* [ ] What is the assumption? An instance of this `ContainerInspector` can inspect only `K` or `K` and all the derived kinds? State it.
* [ ] It seems a bit odd that a specific instantiation of a `ContainerInspector<T>` is tied to a `Container` but *not* to a `Kind`. I guess it's part of your goal of not having to create a new class for each kind, but is this idea holding up with the `Kind::verify` approach we're willing to introduce? At very very least I'd like to see some form of assertion (possibly compile time) that the kind *is compatible* with the container, which, according to what we recently discussed, means that the argument of `K::verify` is the same as `Container::value_type`.